bwalgo/bwt.cs

```diff
@@ -0,0 +1,88 @@
+﻿public class BWTalgo
+{
+    public static (string, int) ForwardConversion(string str)
+    {
+        var permutations = new (string, int)[str.Length];
+        var endCharPosition = str.Length;
+        for (int i = 0; i < str.Length; i++)
+        {
+            permutations[i] = (str.Substring(i) + str.Substring(0, i), endCharPosition - i);
+        }
+
+        Array.Sort(permutations, new StringWithEndIndexesCompaper());
+
+        string result = "";
+        int resultEndIndex = 0;
+        for (int i = 0; i < str.Length; i++)
+        {
+            result = result + permutations[i].Item1[^1];
+            if (permutations[i].Item2 == endCharPosition)
+            {
+                resultEndIndex = i + 1;
+            }
+        }
+
+        return (result, resultEndIndex);
+    }
+
+    public static string InverseConversion(string str, int endIndex)
+    {
+        endIndex--;
+        var arrayOfCountCharsBefore = new int[str.Length];
+        Dictionary<char, int> dictionaryOfCharCount = new Dictionary<char, int>();
```

Comment:
```suggestion
        Dictionary<char, int> dictionaryOfCharCount = new();
```
Не пишите в одном объявлении один тип дважды